### PR TITLE
Client identification for different device types

### PR DESCRIPTION
**Pull request by dbischof90**
Proposed pull request, initial commit. Possibility to specify device type identification. Changed in reference implementation, introduced in source file.

### CODE REVIEW

File: src/spirc.rs [hunk 1]
```diff
@@ -59,7 +59,20 @@ fn initial_state() -> State {
...
+        "unknown" => 0,
+        "computer" => 1,
+        "tablet" => 2,
+        "smartphone" => 3,
+        "speaker" => 4,
+        "tv" => 5,
+        "avr" => 6,
+        "stb" => 7,
+        "audiodongle" => 8,
+        _ => 5,
```
Comment (sashahilton00) [inline src/spirc.rs:73]: Can I suggest that we set the default to the more generic `speaker` (4) type, as a TV isn't appropriate in most cases.

**Review by sashahilton00 — COMMENTED** [review]
Minor change suggested with regards to default device type.

**Comment by plietar** [comment]
This should be using an enum to store the device type 
```rust
enum DeviceType {
  Unknown = 0,
  Computer = 1,
  Tablet = 2,
  [...]
}
```

You can then add a `FromStr` and a `Display` implementation for conversion with strings.
You can also use `as i32` to convert to an integer.

Also it needs support in `src/authentication/discovery.rs`

**Comment by sashahilton00** [comment]
@dbischof90 pinging you incase unnoticed.

File: src/spirc.rs (hunk 1 above)
Comment (dbischof90) [reply src/spirc.rs:73]: Yes, will do!

**Comment by sashahilton00** [comment]
How's this looking @dbischof90?

**Pull request closed without merging**